File: src/classifier.py
import anthropic


def classify(content_blocks: list[dict], rules: dict, file_name: str) -> str:
    """Send file content + rules to Claude and return the matched category name."""
    client = anthropic.Anthropic()

    categories_text = "\n".join(
        f"- {cat['name']}: {', '.join(cat.get('hints', []))}"
        for cat in rules["categories"]
    )
    category_names = [cat["name"] for cat in rules["categories"]]

    prompt_block = {
        "type": "text",
        "text": (
            f"Classify this document into one of these categories:\n\n"
            f"{categories_text}\n\n"
            f"If the document doesn't clearly match any category, respond with \"unclassified\".\n\n"
            f"Document filename: {file_name}\n\n"
            f"Respond with ONLY the exact category name, nothing else."
        ),
    }

    response = client.messages.create(
        model="claude-sonnet-4-6-20250514",
        max_tokens=256,
        messages=[{"role": "user", "content": content_blocks + [prompt_block]}],
    )

    result = response.content[0].text.strip()

    # Fuzzy match against known categories
    for name in category_names:
        if result.lower() == name.lower():
            return name
    return "unclassified"

File: server/src/classifier.ts
import Anthropic from "@anthropic-ai/sdk";
import type { RulesConfig } from "../../shared/types.js";

const client = new Anthropic();

type ContentBlock = Anthropic.Messages.ContentBlockParam;

export async function classify(
  contentBlocks: ContentBlock[],
  rules: RulesConfig,
  fileName: string,
): Promise<string> {
  const categoriesText = rules.categories
    .map((cat) => `- ${cat.name}: ${cat.hints.join(", ")}`)
    .join("\n");

  const prompt: ContentBlock = {
    type: "text",
    text: [
      "Classify this document into one of these categories:",
      "",
      categoriesText,
      "",
      'If the document doesn\'t clearly match any category, respond with "unclassified".',
      "",
      `Document filename: ${fileName}`,
      "",
      "Respond with ONLY the exact category name, nothing else.",
    ].join("\n"),
  };

  const response = await client.messages.create({
    model: "claude-sonnet-4-6-20250514",
    max_tokens: 256,
    messages: [{ role: "user", content: [...contentBlocks, prompt] }],
  });

  const result =
    response.content[0].type === "text"
      ? response.content[0].text.trim()
      : "unclassified";

  // Match against known categories (case-insensitive)
  const match = rules.categories.find(
    (cat) => cat.name.toLowerCase() === result.toLowerCase(),
  );
  return match ? match.name : "unclassified";
}
